libbindgen/tests/tests.rs

```diff
@@ -0,0 +1,150 @@
+extern crate diff;
+#[macro_use]
+extern crate lazy_static;
+extern crate libbindgen;
+extern crate tempdir;
+
+use std::env;
+use std::fs;
+use std::io::{BufReader, BufRead, Read};
+use std::path::{Path, PathBuf};
+use tempdir::TempDir;
+
+lazy_static! {
+    static ref TEMP_DIR: PathBuf = TempDir::new("libbindgen-tests")
+        .expect("Couldn't create temporary directory")
+        .into_path();
+}
+
+fn spawn_bindgen(header: &PathBuf, builder: libbindgen::Builder) -> Result<(), ()> {
+    let file_name = header.file_name()
+        .expect("spawn_bindgen expects a file");
+
+    let mut expected = PathBuf::from(header);
+    expected.pop();
+    expected.pop();
+    expected.push("expectations");
+    expected.push("tests");
+    expected.push(file_name);
+    expected.set_extension("rs");
+
+    let output = match builder.generate() {
+        Ok(bindings) => bindings.to_string(),
+        Err(_) => "".to_string(),
+    };
+
+    let mut buffer = String::new();
+    let _ = fs::File::open(&expected)
+        .expect("Couldn't read from expected test output")
+        .read_to_string(&mut buffer)
+        .expect("Couldn't read from expected test output");
```

Comment:
I thinking about 'Authoring New Tests' in CONTRIBUTING.md.

Would it make sense to output the content of 'output' and return an error here instead of a panic?
It would also mean you'd be able to run all your tests, and get 1 failure instead of the test failing to proceed further.